### PR TITLE
EASY-2092: POSTs MAY overwrite existing files

### PR DESCRIPTION
**Pull request by lindareijnhoudt**
Fixes EASY-2029

#### When applied it will
* Remove 'AlreadySubmitted' error message on POST/PUT
* Allow POST to overwrite files

### CODE REVIEW

File: docs/api/api.yml [hunk 1]
```diff
@@ -329,8 +329,6 @@ paths:
           $ref: "#/components/responses/ForbiddenUpdate"
         404:
           $ref: "#/components/responses/NotFound"
-        409:
-          $ref: "#/components/responses/AlreadySubmitted"
```
Comment (jo-pol) [inline docs/api/api.yml]: I guess you dropped it in favour of `ForbiddenStateTransition`

File: docs/api/api.yml [hunk 2]
```diff
@@ -533,8 +530,10 @@ paths:
           $ref: "#/components/responses/ForbiddenUpdate"
         404:
           $ref: "#/components/responses/NotFound"
-        409:
-          $ref: "#components/responses/ConflictAttemptToOverwriteDirectoryWithFile"
+        409.1:
+          $ref: "#/components/responses/ConflictConcurrentPostNotAllowed"
```
Comment (jo-pol) [inline docs/api/api.yml]: Concurrent uploads would cover both types of requests

File: docs/api/api.yml [hunk 3]
```diff
@@ -642,10 +641,6 @@ components:
       description: |
         Forbidden. Illegal state transition. (Only allowed to transition from DRAFT to SUBMITTED and from REJECTED to DRAFT.)
 
-    AlreadySubmitted:
-      description: |
-        Conflict. The deposit already exists in the submit area. Possibly due to a resubmit.
```
Comment (jo-pol) [inline docs/api/api.yml]: Dropped in favour of `ForbiddenUpdate`?

File: docs/api/api.yml [hunk 4]
```diff
@@ -656,7 +651,7 @@ components:
       description: Conflict. The following file(s) already exist on the server
 
     ConflictConcurrentPostNotAllowed:
-      description: Conflict. Concurrent POSTs to the same deposit are not allowed.
+      description: Conflict. Concurrent POSTs and PUTs to the same deposit are not allowed.
```
Comment (jo-pol) [inline docs/api/api.yml]:
```suggestion
      description: Conflict. Concurrent POSTs and PUTs of file uploads to the same deposit are not allowed.
```

File: docs/api/api.yml (hunk 2 above)
Comment (lindareijnhoudt) [inline docs/api/api.yml]:
```suggestion
          $ref: "#/components/responses/ConflictConcurrentUploadsNotAllowed"
```

File: docs/api/api.yml (hunk 4 above)
Comment (lindareijnhoudt) [inline docs/api/api.yml]:
```suggestion
      description: Conflict. Concurrent file uploads to the same deposit are not allowed.
```

File: docs/api/api.yml (hunk 3 above)
Comment (lindareijnhoudt) [reply docs/api/api.yml]: You like this distinction between AlreadySubmitted and ForbiddenUpdate? If we are explicit in the error message, i think they are conveying the same message

File: docs/api/api.yml [hunk 5]
```diff
@@ -656,7 +651,7 @@ components:
       description: Conflict. The following file(s) already exist on the server
 
     ConflictConcurrentPostNotAllowed:
```
Comment (lindareijnhoudt) [inline docs/api/api.yml]:
```suggestion
    ConflictConcurrentUploadsNotAllowed":
```

File: docs/api/api.yml [hunk 6]
```diff
@@ -437,9 +435,7 @@ paths:
           $ref: "#/components/responses/Unauthorized"
         404:
           $ref: "#/components/responses/NotFound"
-        409.1:
-          $ref: "#/components/responses/ConflictOverwritesFilesOnServer"
-        409.2:
+        409:
           $ref: "#/components/responses/ConflictConcurrentPostNotAllowed"
```
Comment (lindareijnhoudt) [inline docs/api/api.yml]:
```suggestion
          $ref: "#/components/responses/ConflictConcurrentUploadsNotAllowed"
```

File: docs/api/api.yml [hunk 7]
```diff
@@ -407,8 +405,8 @@ paths:
         unzipped in `dir_path`. For each file in the ZIP file, the path in the deposit will be `dir_path`/`path_in_zip`.
         The ZIP file may contain `nested ZIP files`, which will **not** be extracted, but stored as ZIP file.
 
-        Concurrent POST calls to the same deposit are not allowed. The result of concurrent PUT and POST calls is
-        undefined.
+        Existing files at the same location will be overwritten. Concurrent POST or PUT calls to the same deposit are not allowed.
```
Comment (janvanmansum) [inline docs/api/api.yml]:
```suggestion
        Existing files at the target location will be overwritten. Concurrent POST or PUT calls to the same deposit are not allowed.
```

File: docs/api/api.yml (hunk 3 above)
Comment (jo-pol) [reply docs/api/api.yml]: I just was surprised as it seemed off topic. But you are right about conveying the same message. Just wanted to be sure

**Pull request merged**